### PR TITLE
[TS-73] ✨ Add storybook component SegmentPicker

### PR DESCRIPTION
**Pull request by wuharry**
## Description:

-add SegmentPicker Component 

## Changes:

-

## Test Scope:

-

## Screenshots:
image

### CODE REVIEW

File: src/stories/SegmentedPicker/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import clsx from "clsx";
+import { SegmentedPickerProps } from "./type";
+
+export const SegmentedPicker: React.FC<SegmentedPickerProps> = ({
+  className,
+  segments,
+  onSelect,
+}) => {
+  const [selectedSegment, setSelectedSegment] = useState(segments[0]);
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: please provide a props value for have a default selected

File: src/stories/SegmentedPicker/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import clsx from "clsx";
+import { SegmentedPickerProps } from "./type";
+
+export const SegmentedPicker: React.FC<SegmentedPickerProps> = ({
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: please adjust to import FC directly

File: src/stories/SegmentedPicker/type.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export interface SegmentedPickerProps {
+  className?: string;
+  segments: string[];
+  onSelect: (segment: string) => void;
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/type.ts]: please setting onSelect accept undefined `?`

File: src/stories/SegmentedPicker/index.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        className
+      )}
+    >
+      {segments.map((segment) => (
+        <div
+          className={clsx(
+            "flex flex-1 items-center justify-center",
+            "relative gap-1",
+            segment !== selectedSegment &&
+              `
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: please don't use ` when you do not use variable inner

File: src/stories/SegmentedPicker/type.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export interface SegmentedPickerProps {
+  className?: string;
+  segments: string[];
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/type.ts]: please change type from string to { label: string, value: T}, and use label to display, use value to mapping item

File: src/stories/SegmentedPicker/index.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+              segment === selectedSegment
+                ? "bg-white shadow-xl "
+                : "bg-transparent",
+              "flex flex-1 items-center justify-center",
+              "rounded-lg transition-colors duration-200",
+              "z-10"
+            )}
+            onClick={() => onSegmentClick(segment)}
+          >
+            <span
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: please don't use span, use other typography storybook component

File: src/stories/SegmentedPicker/index.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { SegmentedPicker } from "."; // Adjust the import if the file path is different
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.spec.tsx]: remove remark

File: src/stories/SegmentedPicker/index.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.spec.tsx]: remove import react

File: src/stories/SegmentedPicker/index.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+import { render, screen, fireEvent } from "@testing-library/react";
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.spec.tsx]: remove import screen, if you want to use `screen.getByText`, you can get from  `const {getByText} = render(<SegmentedPicker {...defaultProps} />);`

File: src/stories/SegmentedPicker/index.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+import "@testing-library/jest-dom";
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.spec.tsx]: remove this import

**Review by HongLinLiao — CHANGES_REQUESTED** [review]
please fix theres

File: src/stories/SegmentedPicker/index.tsx (hunk 1 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:
  const [selectedSegment, setSelectedSegment] = useState<Segment<any> | null>(
    segments[0].value !== undefined ? segments[0].value : null
  );

File: src/stories/SegmentedPicker/index.tsx (hunk 2 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:export const SegmentedPicker: FC<SegmentedPickerProps<any>> = ({

File: src/stories/SegmentedPicker/type.ts (hunk 5 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/type.ts]: change:add 
 interface Segment<T> {
  label: string;
  value: T;
}
and replace SegmentedPickerProps's segments

File: src/stories/SegmentedPicker/index.tsx (hunk 4 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:replace `` to ""

File: src/stories/SegmentedPicker/index.tsx (hunk 6 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:replace span to Text Component

File: src/stories/SegmentedPicker/type.ts (hunk 3 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/type.ts]: change:  onSelect: (selectedSegment: T) => void;

File: src/stories/SegmentedPicker/index.spec.tsx (hunk 9 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.spec.tsx]: change:
const { getByText } = render(<SegmentedPicker {...defaultProps} />);

    segmentsArray.forEach((segment) => {
      expect(getByText(segment.label)).toBeInTheDocument();
    });

File: src/stories/SegmentedPicker/index.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
+import { FC, useState } from "react";
+import clsx from "clsx";
+import Text from "../Typography/Text";
+import { SegmentedPickerProps, Segment } from "./type";
+
+export const SegmentedPicker: FC<SegmentedPickerProps<any>> = ({
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: please change to T not use any

File: src/stories/SegmentedPicker/index.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  className,
+  segments,
+  onSelect,
+}) => {
+  const [selectedSegment, setSelectedSegment] = useState<Segment<any> | null>(
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: please provide a prop named value，and if provide value，change this useState default value

File: src/stories/SegmentedPicker/type.ts [hunk 13]
```diff
@@ -0,0 +1,10 @@
+export interface Segment<T> {
+  label: string;
+  value: T;
+}
+
+export interface SegmentedPickerProps<T> {
+  className?: string;
+  segments: Segment<T>[];
+  onSelect: (selectedSegment: T) => void;
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/type.ts]: please set onSelect accept undefined

File: src/stories/SegmentedPicker/index.tsx [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+              "after:left-[-8px]",
+              "after:top-2",
+              "after:w-[1px] after:border-x-[0.5px]",
+              "after:content-['']"
+            )}
+            key={`${segment.value}-${index}`}
+          >
+            <button
+              className={clsx(
+                isSelected ? "bg-white shadow-xl " : "bg-transparent",
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx:50]: why should need `bg-transparent` here?

File: src/stories/SegmentedPicker/index.tsx (hunk 11 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:
export const SegmentedPicker = <T,>({
  className,
  segments,
  value,
  onSelect,
}: SegmentedPickerProps<T>) => { ...

File: src/stories/SegmentedPicker/index.tsx (hunk 12 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:
  const [selectedSegment, setSelectedSegment] = useState<Segment<T> | null>(
    segments.find((segment) => segment.value === value) || null
  );

File: src/stories/SegmentedPicker/type.ts (hunk 13 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/type.ts]: change:  onSelect: (selectedSegment: T | undefined) => void;

File: src/stories/SegmentedPicker/index.tsx (hunk 14 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx:50]: 我想加當切換到其他的segment會比較絲滑的效果

File: src/stories/SegmentedPicker/index.tsx [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+import { SegmentedPickerProps, Segment } from "./type";
+
+export const SegmentedPicker = <T,>({
+  className,
+  segments,
+  value,
+  onSelect,
+}: SegmentedPickerProps<T>) => {
+  const [selectedSegment, setSelectedSegment] = useState<Segment<T> | null>(
+    segments.find((segment) => segment.value === value) || null
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: if not provide value, set default as segments first item value

File: src/stories/SegmentedPicker/type.ts [hunk 16]
```diff
@@ -0,0 +1,11 @@
+export interface Segment<T> {
+  label: string;
+  value: T;
+}
+
+export interface SegmentedPickerProps<T> {
+  className?: string;
+  segments: Segment<T>[];
+  value?: T;
+  onSelect: (selectedSegment: T | undefined) => void;
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/type.ts]: 不是把參數設成可接受undefined，而是onSelect可以不用提供
`onSelect?: (selectedSegment: T) => void;`

File: src/stories/SegmentedPicker/index.tsx [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+import Text from "../Typography/Text";
+import { SegmentedPickerProps, Segment } from "./type";
+
+export const SegmentedPicker = <T,>({
+  className,
+  segments,
+  value,
+  onSelect,
+}: SegmentedPickerProps<T>) => {
+  const [selectedSegment, setSelectedSegment] = useState<Segment<T> | null>(
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: not accept null

File: src/stories/SegmentedPicker/index.tsx (hunk 15 above)
Comment (HongLinLiao) [reply src/stories/SegmentedPicker/index.tsx]:
```
const [selectedSegment, setSelectedSegment] = useState<T>(
  segments.find((segment) => segment.value === value)?.value ?? segments[0].value
 );
 ```

File: src/stories/SegmentedPicker/index.tsx [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+    segments.find((segment) => segment.value === value) || null
+  );
+
+  const onSegmentClick = (segment: Segment<T>) => {
+    setSelectedSegment(segment);
+    onSelect(segment.value);
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: onSelect?.(segment.value);

File: src/stories/SegmentedPicker/index.tsx [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+  segments,
+  value,
+  onSelect,
+}: SegmentedPickerProps<T>) => {
+  const [selectedSegment, setSelectedSegment] = useState<Segment<T> | null>(
+    segments.find((segment) => segment.value === value) || null
+  );
+
+  const onSegmentClick = (segment: Segment<T>) => {
+    setSelectedSegment(segment);
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: setSelectedSegment(segment.value);

File: src/stories/SegmentedPicker/index.tsx [hunk 20]
```diff
@@ -0,0 +1,65 @@
+import { FC, useState } from "react";
```
Comment (HongLinLiao) [inline src/stories/SegmentedPicker/index.tsx]: remove FC

Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:import { useState } from "react";

File: src/stories/SegmentedPicker/index.tsx (hunk 19 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:
setSelectedSegment(segment.value);

File: src/stories/SegmentedPicker/index.tsx (hunk 18 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:
 const onSegmentClick = (segment: Segment<T>) => {
    setSelectedSegment(segment.value);
    onSelect?.(segment.value);
  };

File: src/stories/SegmentedPicker/index.tsx (hunk 17 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:
  const [selectedSegment, setSelectedSegment] = useState<T>(
    segments.find((segment) => segment.value === value)?.value ||
      segments[0].value
  );

File: src/stories/SegmentedPicker/type.ts (hunk 16 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/type.ts]: change:
export interface SegmentedPickerProps<T> {
  className?: string;
  segments: Segment<T>[];
  value?: T;
  onSelect?: (selectedSegment: T) => void;
}

File: src/stories/SegmentedPicker/index.tsx (hunk 15 above)
Comment (wuharry) [reply src/stories/SegmentedPicker/index.tsx]: change:
  const [selectedSegment, setSelectedSegment] = useState<T>(
    segments.find((segment) => segment.value === value)?.value ||
      segments[0].value
  );

**Review by HongLinLiao — APPROVED** [review]

**Pull request merged**